Platform/CommonBoardPkg/Library/BoardSupportLib/BoardSupportLib.c

```diff
@@ -78,48 +79,50 @@ FillBootOptionListFromCfgData (
       break;
     }
 
-    if (BootOptionCfgData->ImageType == IMAGE_TYPE_NOT_USED) {
+    if ((BootOptionCfgData->ImageType == IMAGE_TYPE_NOT_USED) && (BootOptionCfgData->PreOsImageType == PREIMAGE_TYPE_NOT_USED)) {
       continue;
     }
 
-    UpdateFlag = 0;
     if (BootOptionCfgData->ImageType == IMAGE_TYPE_ADDENDUM) {
-      // This entry is an addendum for previous boot option entry
-      if (ImageIdx < ARRAY_SIZE(BootOption->Image)) {
-        UpdateFlag = 2;
-        BootOption = &OsBootOptionList->OsBootOption[PrevBootOptionIndex];
+      // This entry is an extra image addendum for previous boot option entry
+      BootOption = &OsBootOptionList->OsBootOption[PrevBootOptionIndex];
+      if (BootOptionCfgData->PreOsImageType > 0x3) {
+        continue;
       }
+      ImageIdx = LoadImageTypeExtra0 + BootOptionCfgData->PreOsImageType;
+    } else if (BootOptionCfgData->ImageType == IMAGE_TYPE_NOT_USED) {

```

Comment:
Why NOT_USED is for PreOS ?   It is a little bit confusing to me.

Reply:
Only when ImageType is not used, the PreOsImageType is used as an indication for PreOsImageType.
OK. let me change to use IMAGE_TYPE_ADDENDUM.